target/riscv/cpu_helper.c

```diff
@@ -1062,41 +1154,46 @@ void riscv_cpu_do_interrupt(CPUState *cs)
                 htval = env->guest_phys_fault_addr;
             }
         }
+        riscv_log_instr_csr_changed(env, CSR_HSTATUS);
 
         s = env->mstatus;
         s = set_field(s, MSTATUS_SPIE, env->priv_ver >= PRIV_VERSION_1_10_0 ?
             get_field(s, MSTATUS_SIE) : get_field(s, MSTATUS_UIE << env->priv));
         s = set_field(s, MSTATUS_SPP, env->priv);
         s = set_field(s, MSTATUS_SIE, 0);
         env->mstatus = s;
+        riscv_log_instr_csr_changed(env, CSR_MSTATUS);
         env->scause = cause | ((target_ulong)async << (TARGET_LONG_BITS - 1));
-        COPY_SPECIAL_REG(env, sepc, SEPCC, pc, PCC);
+        riscv_log_instr_csr_changed(env, CSR_SCAUSE);
+#ifdef TARGET_CHERI
+        env->SEPCC = env->PCC;
+        riscv_log_instr_scr_changed(env, CheriSCR_SEPCC);
```

Comment:
Can we include that in the COPY_SPECIAL_REG macro?

Reply:
Good point